ts/src/pro/poloniex.ts

```diff
@@ -0,0 +1,970 @@
+        /**
+         * @method
+         * @name poloniex#watchOHLCV
+         * @description watches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+         * @see https://docs.poloniex.com/#public-channels-market-data-candlesticks
+         * @param {string} symbol unified symbol of the market to fetch OHLCV data for
+         * @param {string} timeframe the length of time each candle represents
+         * @param {int|undefined} since timestamp in ms of the earliest candle to fetch
+         * @param {int|undefined} limit the maximum amount of candles to fetch
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns [[int]] A list of candles ordered as timestamp, open, high, low, close, volume
+         */
+        const channels = this.options['ohlcvChannels'];
+        const channel = this.safeString (channels, timeframe);
+        if (channel === undefined) {
+            throw new BadRequest (this.id + ' watchOHLCV cannot take a timeframe of ' + timeframe);
+        }
+        const ohlcv = await this.subscribe (channel, false, [ symbol ], params);
+        if (this.newUpdates) {
+            limit = ohlcv.getLimit (symbol, limit);
+        }
+        return this.filterBySinceLimit (ohlcv, since, limit, 0, true);
+    }
+
+    async watchTicker (symbol, params = {}) {
+        /**
+         * @method
+         * @name poloniex#watchTicker
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @see https://docs.poloniex.com/#public-channels-market-data-ticker
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        const name = 'ticker';
+        return await this.subscribe (name, false, [ symbol ], params);
+    }
+
+    async watchTickers (symbols = undefined, params = {}) {
+        /**
+         * @method
+         * @name poloniex#watchTicker
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @see https://docs.poloniex.com/#public-channels-market-data-ticker
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        const name = 'ticker';
+        return await this.subscribe (name, false, symbols, params);
+    }
```

Comment:
Take a look at binances implementation. Here you need to check for newUpdates, filter the tickers by ones subscribed to and only check return if there is a ticker to return